cpp/arcticdb/processing/operation_dispatch_binary.hpp

```diff
@@ -273,20 +282,25 @@ VariantData visit_binary_comparator(const VariantData& left, const VariantData&
     }, left, right);
 }
 
+template <typename Func>
+inline std::string binary_operation_column_name(std::string_view left_column, Func& func, std::string_view right_column) {
```

Comment:
The proper thing is to use universal reference
```suggestion
inline std::string binary_operation_column_name(std::string_view left_column, Func&& func, std::string_view right_column) {
```

Reply:
William had to make some changes in these files to get everything working with C++17 again, if you haven't already done so a rebase is a good idea

Reply:
> William had to make some changes in these files to get everything working with C++17 again, if you haven't already done so a rebase is a good idea

@alexowens90, I already did that. You can see [here](https://github.com/man-group/ArcticDB/commits/bug/query_type_mismatch_message/) that William's C++17 revert is included in this branch.

Reply:
Fixed in [a907db9](https://github.com/man-group/ArcticDB/pull/1557/commits/a907db94c4366371434d048c630f796fcbc7a60e)